src/qtouch/qtouch.c

```diff
@@ -533,21 +551,49 @@ uint16_t qtouch_get_sensor_node_signal(uint16_t sensor_node)
     return (ptc_qtlib_node_stat1[sensor_node].node_acq_signals);
 }
 
+void qtouch_update_sensor_node_signal(uint16_t sensor_node, uint16_t new_signal)
+{
+    ptc_qtlib_node_stat1[sensor_node].node_acq_signals = new_signal;
+}
+
 uint16_t qtouch_get_sensor_node_reference(uint16_t sensor_node)
 {
     return (qtlib_key_data_set1[sensor_node].channel_reference);
 }
 
+void qtouch_update_sensor_node_reference(uint16_t sensor_node, uint16_t new_reference)
+{
+    qtlib_key_data_set1[sensor_node].channel_reference = new_reference;
+}
+
 uint16_t qtouch_get_sensor_cc_val(uint16_t sensor_node)
 {
     return (ptc_qtlib_node_stat1[sensor_node].node_comp_caps);
 }
 
+void qtouch_update_sensor_cc_val(uint16_t sensor_node, uint16_t new_cc_value)
+{
+    ptc_qtlib_node_stat1[sensor_node].node_comp_caps = new_cc_value;
+}
+
 uint8_t qtouch_get_sensor_state(uint16_t sensor_node)
 {
     return (qtlib_key_set1.qtm_touch_key_data[sensor_node].sensor_state);
 }
 
+void qtouch_update_sensor_state(uint16_t sensor_node, uint8_t new_state)
+{
+    qtlib_key_set1.qtm_touch_key_data[sensor_node].sensor_state = new_state;
+}
+
+void qtouch_calibrate_node(uint16_t sensor_node)
```

Comment:
is our own `qtouch_force_calibrate()` still doing what it is supposed to do?

Reply:
I tried interacting during the regular functions like unlocking - trinary input. Worked as before, seemingly no change at all.

Reply:
but I suspect lots of things have changed. it is very possible not only could drop our force_calibrate func with this qtouch upgrade, the custom filtering and scroller funcs might go away too. but it would require some testing and for sure tweaking the sensor values.

Reply:
For context: if you plug in the device while holding the touch sensors, the initial calibration will be off and you touch is broken (the base level is off, so it wouldn't register any more touches after).

qtouch_force_calibrate() resets the base level and is called during the orientation screen, which fixed that issue.

Reply:
Yeah, I imagined something like that from [the comment](https://github.com/digitalbitbox/bitbox02-firmware/blob/27a37d3/src/qtouch/qtouch.c#L312-L318). I was actually more concerned with the custom [qtouch_get_sensor_node_signal_filtered](https://github.com/digitalbitbox/bitbox02-firmware/blob/27a37d3/src/qtouch/qtouch.c#L312-L318) and [qtouch_process_scroller_positions](https://github.com/digitalbitbox/bitbox02-firmware/blob/27a37d3/src/qtouch/qtouch.c#L312-L318) - those are much more involved.

They do say something about "scroller bug fixes" in https://microchipdeveloper.com/touch:release-notes but it is obviously unknown what they did exactly. But it could eliminate the need for the custom scroller although I wouldn't do it here. Some time later on, unless we manage to get rid of qtouch :)

Reply:
i tested that calibration still works